src/models/stores/sorted-documents.ts

```diff
@@ -218,6 +218,57 @@ export class SortedDocuments {
 
     const sortedSectionLabels = ["Bookmarked", "Not Bookmarked"];
     return sortedSectionLabels.filter(label => documentMap.has(label)).map(label => documentMap.get(label));
+  }
+
+  //**************************************** Sort By Tools ****************************************
+
+  get sortByTools(): SortedDocument[] {
+    const tileTypeToDocumentsMap: Record<string, DocumentModelType[]> = {};
 
+    this.filteredDocsByType.forEach((doc) => {
+      const tilesByTypeMap = doc.content?.getAllTilesByType();// Type is Record<string, string[]>
+
+      if (tilesByTypeMap) {
+        const tileTypes = Object.keys(tilesByTypeMap);
+        const nonPlaceholderTiles = tileTypes.filter(type => type !== "Placeholder");
+
+        // If a document only has "Placeholder" tiles or no tiles, treat it as "No Tools"
+        if (nonPlaceholderTiles.length === 0) {
+          if (!tileTypeToDocumentsMap["No Tools"]) {
+            tileTypeToDocumentsMap["No Tools"] = [];
+          }
+          tileTypeToDocumentsMap["No Tools"].push(doc);
```

Comment:
This chunk of code is repeated 3 times. 

I'd recommend adding a function after `const tilesByTypeMap = ...` like:
```
const addDocByType = (doc, type) => {
  if (!tileTypeToDocumentsMap[type]) {
    tileTypeToDocumentsMap[type] = [];
  }
  tileTypeToDocumentsMap[type].push(doc);
};
```
Then you can replace each repeated block with:
`addDocByType(doc, "No Tools")` or `addDocByType(doc, tileType)`

Reply:
I think you meant to say add it below `tileTypeToDocumentsMap` and not under `tilesByTypeMap` otherwise that'd be in the `forEach` and it'd get declared multiple times.